### PR TITLE
HMAN - 27

### PR DESCRIPTION
**Pull request by padmapriyanalam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMAN-27

### Change description ###

Insert "Hearing Request" data in the Database

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by RebeccaBaker — COMMENTED** [review]
some initial comments

File: src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java [hunk 1]
```diff
@@ -48,5 +61,112 @@ void shouldPassWithValidHearingId() {
...
+        partyDetails.get(0).setOrganisationDetails(TestingUtil.organisationDetails());
+        hearingRequest.setPartyDetails(partyDetails);
+        Exception exception = assertThrows(BadRequestException.class, () -> {
+            hearingManagementService.saveHearingRequest(hearingRequest);
+        });
+        assertEquals("Either Individual or Organisation details should be present", exception.getMessage());
+    }
+
+    @Test
+    void shouldPassWithParty_Details_InValid_Dow_details_Present() {
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java:138]: changed

File: src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java [hunk 2]
```diff
@@ -48,5 +61,112 @@ void shouldPassWithValidHearingId() {
...
+        partyDetails.get(0).setUnavailabilityDow(new ArrayList<>());
+        hearingRequest.setPartyDetails(partyDetails);
+        Exception exception = assertThrows(BadRequestException.class, () -> {
+            hearingManagementService.saveHearingRequest(hearingRequest);
+        });
+        assertEquals("Unavailability DOW details should be present", exception.getMessage());
+    }
+
+    @Test
+    void shouldPassWithParty_Details_InValid_UnavailabilityRange_details_Present() {
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/reform/hmc/service/HearingManagementServiceTest.java:155]: changed

File: src/test/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementControllerTest.java [hunk 3]
```diff
@@ -46,4 +48,46 @@ void shouldReturn200_whenRequestIdIsValid() {
...
+    @Test
+    void shouldReturn202_whenHearingRequestData() {
+        HearingRequest hearingRequest = new HearingRequest();
+        hearingRequest.setRequestDetails(TestingUtil.requestDetails());
+        hearingRequest.setHearingDetails(TestingUtil.hearingDetails());
+        hearingRequest.setCaseDetails(TestingUtil.caseDetails());
+        HearingManagementController controller = new HearingManagementController(hearingManagementService);
+        controller.saveHearing(hearingRequest);
+        verify(hearingManagementService, times(1)).saveHearingRequest(any());
+    }
```
[the remark this answers is not in the document]
Comment (padmapriyanalam) [reply src/test/java/uk/gov/hmcts/reform/hmc/controllers/HearingManagementControllerTest.java:91]: these test different mandatory fields in the request object request details, case details, hearing details

**Review by RebeccaBaker — APPROVED** [review]

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**